syntax = "proto3";

package common;

option go_package = "github.com/sameersah/auditchain/proto_gen/common";
option java_package = "com.codecatalyst.auditchain.proto.common";
option java_outer_classname = "CommonProto";

enum AccessType {
  UNKNOWN = 0;
  READ = 1;
  WRITE = 2;
  UPDATE = 3;
  DELETE = 4;
}

message FileInfo {
  string file_id = 1;
  string file_name = 2;
}

message UserInfo {
  string user_id = 1;
  string user_name = 2;
}

message FileAudit {
  string req_id = 1;
  FileInfo file_info = 2;
  UserInfo user_info = 3;
  AccessType access_type = 4;
  int64 timestamp = 5;

  string signature = 6;     // RSA signature (hex/base64)
  string public_key = 7;    // PEM-encoded public key
}